examples/chainermn/cifar/train_cifar.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+
+import chainermn
+import chainer
+import chainer.links as L
+from chainer import training
+from chainer.training import extensions
+
+from chainer.datasets import get_cifar10
+from chainer.datasets import get_cifar100
+
+import models.VGG
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Chainer CIFAR example:')
+    parser.add_argument('--dataset', '-d', default='cifar10',
+                        help='The dataset to use: cifar10 or cifar100')
+    parser.add_argument('--batchsize', '-b', type=int, default=64,
+                        help='Number of images per GPU in a mini-batch')
+    parser.add_argument('--learnrate', '-l', type=float, default=0.05,
+                        help='Learning rate for SGD')
+    parser.add_argument('--epoch', '-e', type=int, default=300,
+                        help='Number of sweeps over the dataset to train')
+    parser.add_argument('--gpu', '-g', action='store_true',
+                        help='Use GPU (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Directory to output the result')
+    parser.add_argument('--resume', '-r', default='',
+                        help='Resume the training from snapshot')
+    parser.add_argument('--communicator', type=str,
+                        default='hierarchical', help='Type of communicator')
```

Comment:
Please use `flat` as the default value.

Reply:
Remove `type=str.`

Reply:
I meant to write `default='flat'`, and I guess `type=str` can be kept :)

Reply:
I misunderstood. I changed it.